ai-ml/trainium-inferentia/main.tf

```diff
@@ -48,4 +56,20 @@ locals {
     Blueprint  = local.name
     GithubRepo = "github.com/awslabs/data-on-eks"
   }
+} */
+
+data "external" "eks_azs" {
+  program = ["bash", "${path.module}/get_eks_azs.sh"]
+}
```

Comment:
Is the inclusion of this shell script essential? While it's clear that `az3` and `az4` in `us-west-2` are required, this setup might not hold for different regions, causing users to adjust the script accordingly.

My suggestion is to introduce a new `azs` variable immediately after the region variable, where you can directly specify the availability zones for the Trn1 instances.

To guide users, we should detail this requirement on our website and in the blog post, emphasizing the need to update the availability zones in tandem with the region within the variables prior to deploying the solution.

Incorporating a link to the `Trn1` node availability regions in our documentation would be beneficial for users.

Reply:
Exporting the region variable and running that script as part of terraform installation is ideal than asking users to specify Availability zones.